src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -483,13 +485,10 @@ private void OnStateChanged(object sender, RachisConsensus.StateTransition state
                 }
 
                 if (state.To == RachisState.LeaderElect  || state.To == RachisState.Leader)
-                {
-                    _engine.CurrentLeader.OnNodeStatusChange += OnTopologyChanged;
-                } else if (state.From == RachisState.LeaderElect || state.From == RachisState.Leader)
                 {
                     _engine.CurrentLeader.OnNodeStatusChange -= OnTopologyChanged;
-                }
-
+                    _engine.CurrentLeader.OnNodeStatusChange += OnTopologyChanged;
```

Comment:
huh? You unregister and register again immediately?

Reply:
To ensure that we have registered the event only once.